### PR TITLE
refactor edge metamanager: introduce processors to distinguish messages

### LINKED ISSUE #3070: Refactor edge metamanager
**What would you like to be added/modified**:
Refactor edgecore metamanager
introduce a structure that contains resourceType and operation like below, introduce a global map value `processors` to contain different processors,

So we can distinguish or dispatch messages by resourceType and Operation to different processors to process messages.  and we can process message like function `func Process(message model.Message)` just as belows.

So when we add modules to process message, we just need to add a new processor to the global map `processors`, but not need to intrusively modify in the function `func (m *metaManager) process(message model.Message)` 
```
var (
	processors = make(map[queryKey]Processor)
)

type queryKey struct {
	resType   string
	operation string
}

type Processor interface {
	Process(message model.Message)
}

func Process(message model.Message) {

	op := message.GetOperation()
	_, resType, _ := parseResource(message.GetResource())
	p := processors[queryKey{resType: resType, operation: op}]
	if resType != "" && p == nil {
		p = processors[queryKey{operation: op}]
	}
	if p != nil {
		p.Process(message)
	} else {
		klog.Errorf("No processor to process message,op:%v,resType:%v", op, resType)
	}
}
```

https://github.com/kubeedge/kubeedge/blob/9abf3aa6ee4df0a0746acd8507e1c9b18af59eb6/edge/pkg/metamanager/process.go#L431-L460
**Why is this needed**:

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: gy95 <guoyao17@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
introduce the processors, and use operationType and resourceType of message to distinguish, so if we want to add more  function modules to add functions of processing more different msgs, we only need to add a new processor, and don't need to modify the below codes intrusively.
https://github.com/kubeedge/kubeedge/blob/c2f3b2af1d59fdfd39ca6944d292eda9b2d1aa7f/edge/pkg/metamanager/process.go#L431-L460
**Which issue(s) this PR fixes**:

Fixes #3070

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/common/modules/names.go [hunk 1]
```diff
@@ -11,4 +11,6 @@ const (
 	EdgeStreamModuleName = "edgestream"
 	// DeviceTwinModuleName name
 	DeviceTwinModuleName = "twin"
+	// MetaManagerModuleName name
+	MetaManagerModuleName = "metaManager"
```
Comment (zc2638) [inline edge/pkg/common/modules/names.go:15]: `EdgeStreamModuleName = "edgestream"`
Uniform case is recommended.

File: edge/pkg/metamanager/processor/common_processor.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+type insertProcessor struct{}
+
+type updateProcessor struct{}
+
+type deleteProcessor struct{}
+
+type queryProcessor struct{}
+
+type responseProcessor struct{}
```
Comment (zc2638) [inline edge/pkg/metamanager/processor/common_processor.go:42]: It is recommended that the method follow the `struct` to increase readability

Comment (gy95) [reply edge/pkg/metamanager/processor/common_processor.go]: done

File: edge/pkg/common/modules/names.go (hunk 1 above)
Comment (gy95) [reply edge/pkg/common/modules/names.go:15]: we can open another PR to fix it, need to modify all "metaManager" related codes, and move this string literal to the common modules just as processor also need to use this module name. If not, compile will be failed due to `import cycle not allowed `

Comment (zc2638) [reply edge/pkg/common/modules/names.go:15]: okey.

**Comment by gy95** [comment]
ping @fisherxu PTAL, review please. 😃

**Pull request closed without merging**